src/registrar/models/user.py

```diff
@@ -172,6 +252,49 @@ def get_formatted_name(self):
     def has_contact_info(self):
         return bool(self.title or self.email or self.phone)
 
+    def _get_portfolio_permissions(self):
+        """
+        Retrieve the permissions for the user's portfolio roles.
+        """
+        portfolio_permissions = set()  # Use a set to avoid duplicate permissions
+
+        if self.portfolio_roles:
+            for role in self.portfolio_roles:
+                if role in self.PORTFOLIO_ROLE_PERMISSIONS:
+                    portfolio_permissions.update(self.PORTFOLIO_ROLE_PERMISSIONS[role])
+        if self.portfolio_additional_permissions:
+            portfolio_permissions.update(self.portfolio_additional_permissions)
+        return list(portfolio_permissions)  # Convert back to list if necessary
+
+    def _has_portfolio_permission(self, portfolio_permission):
+        """The views should only call this function when testing for perms and not rely on roles."""
+
+        # EDIT_DOMAINS === user is a manager on a domain (has UserDomainRole)
+        # NOTE: Should we check whether the domain is in the portfolio?
+        if portfolio_permission == self.UserPortfolioPermissionChoices.EDIT_DOMAINS and self.domains.exists():
+            return True
+
+        if not self.portfolio:
+            return False
+
+        portfolio_permissions = self._get_portfolio_permissions()
+
+        return portfolio_permission in portfolio_permissions
+
+    # the methods below are checks for individual portfolio permissions.  they are defined here
+    # to make them easier to call elsewhere throughout the application
+    def has_base_portfolio_permission(self):
+        return self._has_portfolio_permission(User.UserPortfolioPermissionChoices.VIEW_PORTFOLIO)
+
+    def has_domains_portfolio_permission(self):
```

Comment:
this is the most basic permission shared by all types of portfolio users (members, admins, and admin read only).  it is the permission we are checking to determine basic layout of views, whether or not to display portfolio menus.  i think this should remain VIEW_PORTFOLIO.  do you think that has_base_portfolio_permissions is confusing as a name?

Reply:
The base permission one (has base portfolio_permissions) does make sense! I was referring to the domains_portfolio permission which is tied to "view all domains", I do think we will eventually need to differentiate between view_all domains and viewing the domains tab in general as the naming here implies viewing just domains, when really it's the permission to see all domains. That said it can be handled in the follow on ticket and not something I am blocking on